src/process/filter/index.ts

```diff
@@ -40,7 +40,11 @@ export const filterPostProcess: ProcessorFnSync<FilterScope> = (context: FilterC
         if (scope.filter[path]) {
             let value = get(submission?.data, path);
             if (isObject(value) && isObject(scope.filter[path])) {
-                value = {...value, ...scope.filter[path]}
+                if ((value as any).data) {
```

Comment:
This may have a side effect where if any data object is saved that has a "data" property, it will leave ALL other keys alone except "data".  Why can we not also add a conditional here to only do this for the model type "dataObject"?

Reply:
data property will be just added to other properties `value` has. It will not affect other properies object has. It's kinda difficult to check model type, because component is

Reply:
Setting data will not affect other properties, as it's just adding new property to resulting object. Also, checking model type will require some tricks to do, as post processing works per submission, not per component, so it's not possible to get component's mode type directly. Possible solution for that is to refactor filter scope to make component info contain model type. Are you ok with this solution?